aea/aea.py

```diff
@@ -71,14 +71,16 @@ def __init__(
         :return: None
         """
         super().__init__(
-            name=name,
+            identity=identity,
             connections=connections,
             loop=loop,
             timeout=timeout,
             debug=debug,
             programmatic=programmatic,
         )
 
+        self.identity.public_keys = wallet.public_keys
```

Comment:
we set them here as they are derived from the wallet.

Reply:
I feel it is a bit counterintuitive. It means the `Identiy` object gets modified when passed to the constructor. This may lead to weird behaviour of the identity instance, that might be used outside the constructor.

What if the constructor accepts name, and the `Identity` object is instantiated and populated inside the constructor?

Reply:
I don't know how else you suggest we should modify the identity object. The identity object must be modifiable with the details from the wallet. I am not a fan of "name" in the constructor, the reason of introducing `identity` is that it makes it trivial to add credentials to the identity in the future without breaking the agent api. The agent api is bound to be broken otherwise. "name" is too specific, imo only object should be passed to the Agent/AEA

Reply:
We could not modify the identity here and instead pass the details from the wallet to the identity outside the agent, at identity construction time.